# Investigating the Datasets to pick the appropriate one for the CA

## 1. Looking up Population Estimates (Persons in April), PEA01.20231004T131011.csv:

In [1]:
# import pandas data set
import pandas as pd

In [2]:
# Loading the dataset:
df1 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA01.20231004T131011.csv')

In [3]:
# Displaying first 5 rows:
df1.head()

STATISTIC Label  Year     Age Group         Sex  \
0  Population Estimates (Persons in April)  1950  Under 1 year  Both sexes   
1  Population Estimates (Persons in April)  1950  Under 1 year        Male   
2  Population Estimates (Persons in April)  1950  Under 1 year      Female   
3  Population Estimates (Persons in April)  1950   0 - 4 years  Both sexes   
4  Population Estimates (Persons in April)  1950   0 - 4 years        Male   

       UNIT  VALUE  
0  Thousand   61.1  
1  Thousand   31.4  
2  Thousand   29.7  
3  Thousand    NaN  
4  Thousand    NaN

In [4]:
# Looking up distinct values within the df:
def df1_distinct(df1):
    for column in df1:
        unique_values = df1[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df1_distinct(df1)

Unique values for STATISTIC Label:
['Population Estimates (Persons in April)']


Unique values for Year:
[1950 1951 1952 1953 1954 1955 1956 1957 1958 1959 1960 1961 1962 1963
 1964 1965 1966 1967 1968 1969 1970 1971 1972 1973 1974 1975 1976 1977
 1978 1979 1980 1981 1982 1983 1984 1985 1986 1987 1988 1989 1990 1991
 1992 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019
 2020 2021 2022 2023]


Unique values for Age Group:
['Under 1 year' '0 - 4 years' '0 - 14 years' '1 - 4 years' '5 - 9 years'
 '10 - 14 years' '15 - 19 years' '15 - 24 years' '15 years and over'
 '20 - 24 years' '25 - 29 years' '25 - 44 years' '30 - 34 years'
 '35 - 39 years' '40 - 44 years' '45 - 49 years' '45 - 64 years'
 '50 - 54 years' '55 - 59 years' '60 - 64 years' '65 - 69 years'
 '65 years and over' '70 - 74 years' '75 - 79 years' '80 - 84 years'
 '85 years and over' 'All ages']


Unique values for Sex:
['Both sexes' 'Male' 'F

In [5]:
# Getting df info
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5994 entries, 0 to 5993
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  5994 non-null   object 
 1   Year             5994 non-null   int64  
 2   Age Group        5994 non-null   object 
 3   Sex              5994 non-null   object 
 4   UNIT             5994 non-null   object 
 5   VALUE            5880 non-null   float64
dtypes: float64(1), int64(1), object(4)
memory usage: 281.1+ KB


In [6]:
# Describing the df:
df1.describe()

Year        VALUE
count  5994.000000  5880.000000
mean   1986.500000   347.001088
std      21.361791   601.822682
min    1950.000000     5.500000
25%    1968.000000    79.475000
50%    1986.500000   151.700000
75%    2005.000000   307.725000
max    2023.000000  5281.600000

In [7]:
# Counting the NaN values:
df1.isnull().sum()

STATISTIC Label      0
Year                 0
Age Group            0
Sex                  0
UNIT                 0
VALUE              114
dtype: int64

In [8]:
# Display the rows with null values
df1_NaN = df1[df1.isnull().any(axis=1)]
print("Rows with null values:")
df1_NaN.head(10)

Rows with null values:


STATISTIC Label  Year    Age Group         Sex  \
3    Population Estimates (Persons in April)  1950  0 - 4 years  Both sexes   
4    Population Estimates (Persons in April)  1950  0 - 4 years        Male   
5    Population Estimates (Persons in April)  1950  0 - 4 years      Female   
165  Population Estimates (Persons in April)  1952  0 - 4 years  Both sexes   
166  Population Estimates (Persons in April)  1952  0 - 4 years        Male   
167  Population Estimates (Persons in April)  1952  0 - 4 years      Female   
246  Population Estimates (Persons in April)  1953  0 - 4 years  Both sexes   
247  Population Estimates (Persons in April)  1953  0 - 4 years        Male   
248  Population Estimates (Persons in April)  1953  0 - 4 years      Female   
327  Population Estimates (Persons in April)  1954  0 - 4 years  Both sexes   

         UNIT  VALUE  
3    Thousand    NaN  
4    Thousand    NaN  
5    Thousand    NaN  
165  Thousand    NaN  
166  Thousand    NaN  
167  Thousand    NaN  
246  Thousand    NaN  
247  Thousand    NaN  
248  Thousand    NaN  
327  Thousand    NaN

In [9]:
#Function to count the NaN values per category:
def df1_NaN_Count(df1_NaN):
    for column in df1_NaN:
        counts = df1_NaN[column].value_counts()
        print(f"Count {column}:")
        print(counts)
        print("\n")


# Call the function with your DataFrame as an argument
df1_NaN_Count(df1_NaN)

Count STATISTIC Label:
Population Estimates (Persons in April)    114
Name: STATISTIC Label, dtype: int64


Count Year:
1950    3
1984    3
1975    3
1976    3
1977    3
1978    3
1980    3
1982    3
1983    3
1985    3
1952    3
1987    3
1988    3
1989    3
1990    3
1992    3
1993    3
1994    3
1974    3
1973    3
1972    3
1970    3
1953    3
1954    3
1955    3
1956    3
1957    3
1958    3
1959    3
1960    3
1962    3
1963    3
1964    3
1965    3
1967    3
1968    3
1969    3
1995    3
Name: Year, dtype: int64


Count Age Group:
0 - 4 years    114
Name: Age Group, dtype: int64


Count Sex:
Both sexes    38
Male          38
Female        38
Name: Sex, dtype: int64


Count UNIT:
Thousand    114
Name: UNIT, dtype: int64


Count VALUE:
Series([], Name: VALUE, dtype: int64)




Conclusion:

This dataset has 114 missing values for category Age Group 0 - 4 years. It has 5994 entries.

## 2. Looking up Estimated Migration (Persons in April) PEA03.20231004T131049.csv:

In [10]:
# Loading the dataset:
df2 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA03.20231004T131049.csv')


In [11]:
#Displaying first 5 rows:
df2.head()

STATISTIC Label  Year Age Group         Sex  \
0  Estimated Migration (Persons in April)  1987  All ages  Both sexes   
1  Estimated Migration (Persons in April)  1987  All ages  Both sexes   
2  Estimated Migration (Persons in April)  1987  All ages  Both sexes   
3  Estimated Migration (Persons in April)  1987  All ages        Male   
4  Estimated Migration (Persons in April)  1987  All ages        Male   

        Inward or Outward Flow      UNIT  VALUE  
0                Net migration  Thousand  -23.0  
1  Emigrants: All destinations  Thousand   40.2  
2      Immigrants: All origins  Thousand   17.2  
3                Net migration  Thousand    NaN  
4  Emigrants: All destinations  Thousand    NaN

In [12]:
# Looking up distinct values within the df:
def df2_distinct(df2):
    for column in df2:
        unique_values = df2[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df2_distinct(df2)

Unique values for STATISTIC Label:
['Estimated Migration (Persons in April)']


Unique values for Year:
[1987 1988 1989 1990 1991 1992 1993 1994 1995 1996 1997 1998 1999 2000
 2001 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014
 2015 2016 2017 2018 2019 2020 2021 2022 2023]


Unique values for Age Group:
['All ages' '0 - 14 years' '15 - 24 years' '25 - 44 years' '45 - 64 years'
 '65 years and over']


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Inward or Outward Flow:
['Net migration' 'Emigrants: All destinations' 'Immigrants: All origins']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[-2.300e+01  4.020e+01  1.720e+01        nan  2.800e+00  3.100e+00
  1.400e+00  1.300e+00  1.800e+00  2.400e+01  5.100e+00  1.230e+01
  1.900e+00  1.170e+01  3.200e+00  1.180e+01  6.100e+00  6.900e+00
  3.400e+00  4.900e+00  2.700e+00  1.700e+00  1.000e+00  7.000e-01
  8.000e-01  0.000e+00  1.100e+00  5.000e-01  6.000e-01 -4.190e+01
  6.

In [13]:
# Getting df info
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1998 entries, 0 to 1997
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   STATISTIC Label         1998 non-null   object 
 1   Year                    1998 non-null   int64  
 2   Age Group               1998 non-null   object 
 3   Sex                     1998 non-null   object 
 4   Inward or Outward Flow  1998 non-null   object 
 5   UNIT                    1998 non-null   object 
 6   VALUE                   1683 non-null   float64
dtypes: float64(1), int64(1), object(5)
memory usage: 109.4+ KB


In [14]:
# Describing the df:
df2.describe()

Year        VALUE
count  1998.000000  1683.000000
mean   2005.000000    10.856625
std      10.679751    17.070445
min    1987.000000   -43.900000
25%    1996.000000     1.200000
50%    2005.000000     4.500000
75%    2014.000000    14.000000
max    2023.000000   151.100000

In [15]:
# Counting the NaN values:
df2.isnull().sum()

STATISTIC Label             0
Year                        0
Age Group                   0
Sex                         0
Inward or Outward Flow      0
UNIT                        0
VALUE                     315
dtype: int64

In [16]:
# Display the rows with null values
df2_NaN = df2[df2.isnull().any(axis=1)]
print("Rows with null values:")
df2_NaN.head(10)

Rows with null values:


STATISTIC Label  Year      Age Group         Sex  \
3   Estimated Migration (Persons in April)  1987       All ages        Male   
4   Estimated Migration (Persons in April)  1987       All ages        Male   
5   Estimated Migration (Persons in April)  1987       All ages        Male   
6   Estimated Migration (Persons in April)  1987       All ages      Female   
7   Estimated Migration (Persons in April)  1987       All ages      Female   
8   Estimated Migration (Persons in April)  1987       All ages      Female   
9   Estimated Migration (Persons in April)  1987   0 - 14 years  Both sexes   
12  Estimated Migration (Persons in April)  1987   0 - 14 years        Male   
15  Estimated Migration (Persons in April)  1987   0 - 14 years      Female   
18  Estimated Migration (Persons in April)  1987  15 - 24 years  Both sexes   

         Inward or Outward Flow      UNIT  VALUE  
3                 Net migration  Thousand    NaN  
4   Emigrants: All destinations  Thousand    NaN  
5       Immigrants: All origins  Thousand    NaN  
6                 Net migration  Thousand    NaN  
7   Emigrants: All destinations  Thousand    NaN  
8       Immigrants: All origins  Thousand    NaN  
9                 Net migration  Thousand    NaN  
12                Net migration  Thousand    NaN  
15                Net migration  Thousand    NaN  
18                Net migration  Thousand    NaN

In [17]:
#Function to count the NaN values per category:
def df2_NaN_Count(df2_NaN):
    for column in df2_NaN:
        counts = df2_NaN[column].value_counts()
        print(f"Count {column}:")
        print(counts)
        print("\n")


# Call the function with your DataFrame as an argument
df2_NaN_Count(df2_NaN)

Count STATISTIC Label:
Estimated Migration (Persons in April)    315
Name: STATISTIC Label, dtype: int64


Count Year:
1987    21
1988    21
1989    21
1990    21
1991    21
1992    21
1993    21
1994    21
1995    21
1996    21
1997    21
1998    21
1999    21
2000    21
2001    21
Name: Year, dtype: int64


Count Age Group:
All ages             90
0 - 14 years         45
15 - 24 years        45
25 - 44 years        45
45 - 64 years        45
65 years and over    45
Name: Age Group, dtype: int64


Count Sex:
Male          120
Female        120
Both sexes     75
Name: Sex, dtype: int64


Count Inward or Outward Flow:
Net migration                  255
Emigrants: All destinations     30
Immigrants: All origins         30
Name: Inward or Outward Flow, dtype: int64


Count UNIT:
Thousand    315
Name: UNIT, dtype: int64


Count VALUE:
Series([], Name: VALUE, dtype: int64)




Conclusion:

This dataset has 315 missing values spread across 6 Age groups. It has 1998 entries.

## 3. Looking up Estimated Population (Persons in April), PEA04.20231004T131011.csv:

In [18]:
# Loading the dataset:
df3 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA04.20231004T131011.csv')

In [19]:
# Displaying first 5 rows:
df3.head()

STATISTIC Label  Year    Age Group         Sex  \
0  Estimated Population (Persons in April)  2011  0 - 4 years  Both sexes   
1  Estimated Population (Persons in April)  2011  0 - 4 years  Both sexes   
2  Estimated Population (Persons in April)  2011  0 - 4 years  Both sexes   
3  Estimated Population (Persons in April)  2011  0 - 4 years  Both sexes   
4  Estimated Population (Persons in April)  2011  0 - 4 years  Both sexes   

       Region      UNIT  VALUE  
0       State  Thousand  356.0  
1      Border  Thousand   30.7  
2        West  Thousand   32.6  
3    Mid-West  Thousand   35.0  
4  South-East  Thousand   32.0

In [20]:
# Looking up distinct values within the df:
def df1_distinct(df3):
    for column in df3:
        unique_values = df3[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df1_distinct(df3)

Unique values for STATISTIC Label:
['Estimated Population (Persons in April)']


Unique values for Year:
[2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023]


Unique values for Age Group:
['0 - 4 years' '5 - 9 years' '10 - 14 years' '15 - 19 years'
 '20 - 24 years' '25 - 29 years' '30 - 34 years' '35 - 39 years'
 '40 - 44 years' '45 - 49 years' '50 - 54 years' '55 - 59 years'
 '60 - 64 years' '65 - 69 years' '70 - 74 years' '75 - 79 years'
 '80 - 84 years' '85 years and over' 'All ages']


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Region:
['State' 'Border' 'West' 'Mid-West' 'South-East' 'South-West' 'Dublin'
 'Mid-East' 'Midland']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[356.   30.7  32.6 ... 383.8 765.6 391.3]




In [21]:
# Getting df info
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6669 entries, 0 to 6668
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  6669 non-null   object 
 1   Year             6669 non-null   int64  
 2   Age Group        6669 non-null   object 
 3   Sex              6669 non-null   object 
 4   Region           6669 non-null   object 
 5   UNIT             6669 non-null   object 
 6   VALUE            6669 non-null   float64
dtypes: float64(1), int64(1), object(5)
memory usage: 364.8+ KB


In [22]:
# Describing the df:
df3.describe()

Year        VALUE
count  6669.000000  6669.000000
mean   2017.000000    75.670460
std       3.741938   281.512172
min    2011.000000     1.200000
25%    2014.000000    12.500000
50%    2017.000000    20.700000
75%    2020.000000    42.200000
max    2023.000000  5281.600000

In [23]:
# Counting 
df3.isnull().sum()

STATISTIC Label    0
Year               0
Age Group          0
Sex                0
Region             0
UNIT               0
VALUE              0
dtype: int64

Conclusion:

This dataset has no null values however year range goes from 2011 to 2023. It has 6669 entries

## 4. Looking up Estimated Population (Persons in April), PEA07.20231004T131011.csv:

In [24]:
# Loading the dataset:
df4 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA07.20231004T131011.csv')

In [25]:
# Displaying first 5 rows:
df4.head(50)

Statistic Label    Age Group         Sex  \
0   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
1   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
2   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
3   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
4   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
5   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
6   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
7   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
8   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
9   Estimated Population (Persons in April)  0 - 4 years  Both sexes   
10  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
11  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
12  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
13  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
14  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
15  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
16  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
17  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
18  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
19  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
20  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
21  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
22  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
23  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
24  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
25  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
26  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
27  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
28  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
29  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
30  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
31  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
32  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
33  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
34  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
35  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
36  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
37  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
38  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
39  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
40  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
41  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
42  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
43  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
44  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
45  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
46  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
47  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
48  Estimated Population (Persons in April)  0 - 4 years  Both sexes   
49  Estimated Population (Persons in April)  0 - 4 years  Both sexes   

   Regional Authority Area  Year      UNIT  VALUE  
0                    State  1996  Thousand  250.4  
1                    State  1997  Thousand  250.0  
2                    State  1998  Thousand  253.7  
3                    State  1999  Thousand  259.8  
4                    State  2000  Thousand  266.2  
5                    State  2001  Thousand 

In [26]:
# Looking up distinct values within the df:
def df4_distinct(df4):
    for column in df4:
        unique_values = df4[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df4_distinct(df4)

Unique values for Statistic Label:
['Estimated Population (Persons in April)']


Unique values for Age Group:
['0 - 4 years' '5 - 9 years' '10 - 14 years' '15 - 19 years'
 '20 - 24 years' '25 - 29 years' '30 - 34 years' '35 - 39 years'
 '40 - 44 years' '45 - 49 years' '50 - 54 years' '55 - 59 years'
 '60 - 64 years' '65 - 69 years' '70 - 74 years' '75 - 79 years'
 '80 - 84 years' '85 years and over' 'All ages']


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Regional Authority Area:
['State' 'Border' 'Midland' 'West' 'Dublin' 'Mid-East' 'Mid-West'
 'South-East' 'South-West']


Unique values for Year:
[1996 1997 1998 1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009
 2010 2011 2012 2013 2014 2015 2016 2017]


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[250.4 250.  253.7 ... 341.6 343.3 345.3]




In [27]:
# Getting df info
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11286 entries, 0 to 11285
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Statistic Label          11286 non-null  object 
 1   Age Group                11286 non-null  object 
 2   Sex                      11286 non-null  object 
 3   Regional Authority Area  11286 non-null  object 
 4   Year                     11286 non-null  int64  
 5   UNIT                     11286 non-null  object 
 6   VALUE                    11286 non-null  float64
dtypes: float64(1), int64(1), object(5)
memory usage: 617.3+ KB


In [28]:
# Describing the df:
df1.describe()

Year        VALUE
count  5994.000000  5880.000000
mean   1986.500000   347.001088
std      21.361791   601.822682
min    1950.000000     5.500000
25%    1968.000000    79.475000
50%    1986.500000   151.700000
75%    2005.000000   307.725000
max    2023.000000  5281.600000

Conclusion:
    
No NaN values in this dataset, it has 11286 entries.

## 5. Looking up Population estimates from 1926, PEA11.20231004T131042.csv:

In [29]:
# Loading the dataset:
df5 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA11.20231004T131042.csv')

In [30]:
# Displaying first 5 rows:
df5.head()

STATISTIC Label  Year Single Year of Age         Sex  \
0  Population estimates from 1926  1926           All ages  Both sexes   
1  Population estimates from 1926  1926           All ages        Male   
2  Population estimates from 1926  1926           All ages      Female   
3  Population estimates from 1926  1926       Under 1 year  Both sexes   
4  Population estimates from 1926  1926       Under 1 year        Male   

     UNIT    VALUE  
0  Number  2971992  
1  Number  1506889  
2  Number  1465103  
3  Number    55391  
4  Number    28084

In [31]:
# Looking up distinct values within the df:
def df5_distinct(df5):
    for column in df5:
        unique_values = df5[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df5_distinct(df5)

Unique values for STATISTIC Label:
['Population estimates from 1926']


Unique values for Year:
[1926 1936 1946 1951 1961 1966 1971 1979 1981 1986 1991 1996 1997 1998
 1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011 2012
 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023]


Unique values for Single Year of Age:
['All ages' 'Under 1 year' '1 year' '2 years' '3 years' '4 years'
 '5 years' '6 years' '7 years' '8 years' '9 years' '10 years' '11 years'
 '12 years' '13 years' '14 years' '15 years' '16 years' '17 years'
 '18 years' '19 years' '20 years' '21 years' '22 years' '23 years'
 '24 years' '25 years' '26 years' '27 years' '28 years' '29 years'
 '30 years' '31 years' '32 years' '33 years' '34 years' '35 years'
 '36 years' '37 years' '38 years' '39 years' '40 years' '41 years'
 '42 years' '43 years' '44 years' '45 years' '46 years' '47 years'
 '48 years' '49 years' '50 years' '51 years' '52 years' '53 years'
 '54 years' '55 years' '56 years' '57 years' '58 years' '

In [32]:
# Getting df info
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11817 entries, 0 to 11816
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   STATISTIC Label     11817 non-null  object
 1   Year                11817 non-null  int64 
 2   Single Year of Age  11817 non-null  object
 3   Sex                 11817 non-null  object
 4   UNIT                11817 non-null  object
 5   VALUE               11817 non-null  int64 
dtypes: int64(2), object(4)
memory usage: 554.0+ KB


In [33]:
# Describing the df:
df5.describe()

Year         VALUE
count  11817.000000  1.181700e+04
mean    1996.410256  5.352461e+04
std       24.463165  2.867754e+05
min     1926.000000  6.000000e+00
25%     1986.000000  1.242700e+04
50%     2004.000000  2.667500e+04
75%     2014.000000  3.557700e+04
max     2023.000000  5.281612e+06

Conclusion:
    
No NaN values in for this df, it has 11286 entries.

## 6. Looking up Anual Population Change, PEA15.20231004T131032.csv:

In [34]:
# Loading the dataset:
df6 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA15.20231004T131032.csv')

In [35]:
# Displaying first 5 rows:
df6.head()

STATISTIC Label  Year         Component      UNIT  VALUE
0  Annual Population Change  1951     Annual births  Thousand    NaN
1  Annual Population Change  1951     Annual deaths  Thousand    NaN
2  Annual Population Change  1951  Natural increase  Thousand   26.6
3  Annual Population Change  1951        Immigrants  Thousand    NaN
4  Annual Population Change  1951         Emigrants  Thousand    NaN

In [36]:
# Looking up distinct values within the df:
def df1_distinct(df6):
    for column in df6:
        unique_values = df6[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df1_distinct(df6)

Unique values for STATISTIC Label:
['Annual Population Change']


Unique values for Year:
[1951 1952 1953 1954 1955 1956 1957 1958 1959 1960 1961 1962 1963 1964
 1965 1966 1967 1968 1969 1970 1971 1972 1973 1974 1975 1976 1977 1978
 1979 1980 1981 1982 1983 1984 1985 1986 1987 1988 1989 1990 1991 1992
 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005 2006
 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020
 2021 2022 2023]


Unique values for Component:
['Annual births' 'Annual deaths' 'Natural increase' 'Immigrants'
 'Emigrants' 'Net migration' 'Population change' 'Population']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[        nan  2.6600e+01 -3.5000e+01 -8.4000e+00  2.9606e+03  2.7300e+01
 -7.7000e+00  2.9529e+03  2.9100e+01 -3.3000e+01 -3.9000e+00  2.9490e+03
  2.8200e+01 -3.6000e+01 -7.8000e+00  2.9412e+03  2.4700e+01 -4.5000e+01
 -2.0300e+01  2.9209e+03  2.5600e+01 -4.8000e+01 -2.2400e+01  2.8985e+03
  2.7800e+01 -4.1000e

In [37]:
# Getting df info
df6.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 584 entries, 0 to 583
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  584 non-null    object 
 1   Year             584 non-null    int64  
 2   Component        584 non-null    object 
 3   UNIT             584 non-null    object 
 4   VALUE            440 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 22.9+ KB


In [38]:
# Describing the df:
df6.describe()

Year        VALUE
count   584.000000   440.000000
mean   1987.000000   633.262273
std      21.089371  1378.671718
min    1951.000000   -58.000000
25%    1969.000000    24.775000
50%    1987.000000    35.550000
75%    2005.000000    70.775000
max    2023.000000  5281.600000

In [39]:
# Counting 
df6.isnull().sum()

STATISTIC Label      0
Year                 0
Component            0
UNIT                 0
VALUE              144
dtype: int64

In [40]:
# Display the rows with null values
df6_NaN = df6[df6.isnull().any(axis=1)]
print("Rows with null values:")
df6_NaN.head(10)

Rows with null values:


STATISTIC Label  Year      Component      UNIT  VALUE
0   Annual Population Change  1951  Annual births  Thousand    NaN
1   Annual Population Change  1951  Annual deaths  Thousand    NaN
3   Annual Population Change  1951     Immigrants  Thousand    NaN
4   Annual Population Change  1951      Emigrants  Thousand    NaN
8   Annual Population Change  1952  Annual births  Thousand    NaN
9   Annual Population Change  1952  Annual deaths  Thousand    NaN
11  Annual Population Change  1952     Immigrants  Thousand    NaN
12  Annual Population Change  1952      Emigrants  Thousand    NaN
16  Annual Population Change  1953  Annual births  Thousand    NaN
17  Annual Population Change  1953  Annual deaths  Thousand    NaN

In [41]:
#Function to count the NaN values per category:
def df6_NaN_Count(df6_NaN):
    for column in df6_NaN:
        counts = df6_NaN[column].value_counts()
        print(f"Count {column}:")
        print(counts)
        print("\n")


# Call the function with your DataFrame as an argument
df6_NaN_Count(df6_NaN)

Count STATISTIC Label:
Annual Population Change    144
Name: STATISTIC Label, dtype: int64


Count Year:
1951    4
1952    4
1971    4
1972    4
1973    4
1974    4
1975    4
1976    4
1977    4
1978    4
1979    4
1980    4
1981    4
1982    4
1983    4
1984    4
1985    4
1970    4
1969    4
1968    4
1959    4
1953    4
1954    4
1955    4
1956    4
1957    4
1958    4
1960    4
1967    4
1961    4
1962    4
1963    4
1964    4
1965    4
1966    4
1986    4
Name: Year, dtype: int64


Count Component:
Annual births    36
Annual deaths    36
Immigrants       36
Emigrants        36
Name: Component, dtype: int64


Count UNIT:
Thousand    144
Name: UNIT, dtype: int64


Count VALUE:
Series([], Name: VALUE, dtype: int64)




In [42]:
df6_unique_count_Year = df6['Year'].nunique()

print(f"Count of years: {df6_unique_count_Year}")

df6_unique_count_Year_NaN = df6_NaN['Year'].nunique()

print(f"Number of years affected by NaN values: {df6_unique_count_Year_NaN}")

print(f"Number of years without NaN values: {df6_unique_count_Year-df6_unique_count_Year_NaN}")

Count of years: 73
Number of years affected by NaN values: 36
Number of years without NaN values: 37


Conclusion:
 
Four categories afected by Nan Values (Annual births, Annual deaths, Immigrants and Emigrants) and 36 years with NaN values.
Total of 584 entries.

## 7. Looking up Estimated Emigration (Persons in April), PEA16.20231004T131017.csv:

In [43]:
# Loading the dataset:
df7 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA16.20231004T131017.csv')

In [44]:
# Displaying first 5 rows:
df7.head()

STATISTIC Label  Year         Sex  \
0  Estimated Emigration (Persons in April)  2006  Both sexes   
1  Estimated Emigration (Persons in April)  2006  Both sexes   
2  Estimated Emigration (Persons in April)  2006  Both sexes   
3  Estimated Emigration (Persons in April)  2006  Both sexes   
4  Estimated Emigration (Persons in April)  2006  Both sexes   

                                         Nationality      UNIT  VALUE  
0                                              Irish  Thousand   15.3  
1                                             UK (1)  Thousand    2.2  
2  EU14 excl Irl (countries in the EU pre 2004 ex...  Thousand    5.1  
3  EU15 to EU27 (accession countries joined post ...  Thousand    7.2  
4                            Other nationalities(17)  Thousand    6.2

In [45]:
# Looking up distinct values within the df:
def df7_distinct(df7):
    for column in df7:
        unique_values = df7[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df7_distinct(df7)

Unique values for STATISTIC Label:
['Estimated Emigration (Persons in April)']


Unique values for Year:
[2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019
 2020 2021 2022]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Nationality:
['Irish' 'UK (1)'
 'EU14 excl Irl (countries in the EU pre 2004 excluding UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)'
 'Other nationalities(17)' 'All nationalities']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[15.3  2.2  5.1  7.2  6.2 36.   8.   1.6  2.5  3.7  2.8 18.7  7.3  0.6
  2.6  3.4 17.3 12.9  8.9 12.6  8.2 46.3  6.1  2.3  4.3  8.4  4.6 25.7
  6.8  1.4  3.6 20.6 13.1  6.  17.2  9.  49.2  7.1  2.  12.4  5.5 29.6
  1.3  4.   4.8  3.5 19.6 19.2  3.9  7.4 30.5 11.  72.  12.3  2.4  1.9
 21.1 41.9  6.9  1.5  9.5  6.7 30.1 28.9  3.  19.   9.3 69.2 16.7  1.7
 12.8  5.4 40.6 12.2  5.  28.7 42.  10.2 13.9  9.9 80.6 24.5  1.8 17.5
  6.6 38.8 49.7 10.7 83.  27.6  4.

In [46]:
# Getting df info
df7.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 306 entries, 0 to 305
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  306 non-null    object 
 1   Year             306 non-null    int64  
 2   Sex              306 non-null    object 
 3   Nationality      306 non-null    object 
 4   UNIT             306 non-null    object 
 5   VALUE            306 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 14.5+ KB


In [47]:
# Describing the df:
df7.describe()

Year       VALUE
count   306.000000  306.000000
mean   2014.000000   14.051307
std       4.907004   16.451076
min    2006.000000    0.600000
25%    2010.000000    3.600000
50%    2014.000000    6.700000
75%    2018.000000   18.650000
max    2022.000000   83.000000

Conclusion:

There are no missing values for this data. Total of 306 entries.

## 8. Looking up Estimated Emigration (Persons in April), PEA17.20231004T131052.csv:

In [48]:
# Loading the dataset:
df8 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA17.20231004T131052.csv')

In [49]:
# Displaying first 5 rows:
df8.head()

STATISTIC Label  Year         Sex  \
0  Estimated Immigration (Persons in April)  1996  Both sexes   
1  Estimated Immigration (Persons in April)  1996  Both sexes   
2  Estimated Immigration (Persons in April)  1996  Both sexes   
3  Estimated Immigration (Persons in April)  1996  Both sexes   
4  Estimated Immigration (Persons in April)  1996  Both sexes   

                                         Nationality      UNIT  VALUE  
0                                              Irish  Thousand   17.7  
1                                             UK (1)  Thousand    8.3  
2  EU14 excl Irl (countries in the EU pre 2004 ex...  Thousand    5.0  
3  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
4                            Other nationalities(17)  Thousand    8.2

In [50]:
# Looking up distinct values within the df:
def df8_distinct(df8):
    for column in df8:
        unique_values = df8[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df8_distinct(df8)

Unique values for STATISTIC Label:
['Estimated Immigration (Persons in April)']


Unique values for Year:
[1996 1997 1998 1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009
 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Nationality:
['Irish' 'UK (1)'
 'EU14 excl Irl (countries in the EU pre 2004 excluding UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)'
 'Other nationalities(17)' 'All nationalities']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[ 17.7   8.3   5.    nan   8.2  39.2   4.3   2.2   4.   18.8   9.5   3.9
   2.8   4.2  20.4  20.8   8.4   5.5   9.7  44.5  10.2   4.6   2.5   4.7
  22.   10.7   3.8   3.1   5.1  22.6  24.3   8.6   6.1   7.   46.   12.3
   2.3   3.7  22.7  11.9   4.4   3.3  23.3  26.7   6.9  48.9  13.7   3.
   3.6  24.5  13.    3.5  24.4  24.8  11.1  52.6  11.8   4.1   6.4  26.2
   4.8  26.4  26.3   9.    6.5  17.3  59.   1

In [51]:
# Getting df info
df8.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 486 entries, 0 to 485
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  486 non-null    object 
 1   Year             486 non-null    int64  
 2   Sex              486 non-null    object 
 3   Nationality      486 non-null    object 
 4   UNIT             486 non-null    object 
 5   VALUE            459 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 22.9+ KB


In [52]:
# Describing the df:
df8.describe()

Year       VALUE
count   486.000000  459.000000
mean   2009.000000   17.263399
std       7.796907   20.004094
min    1996.000000    1.100000
25%    2002.000000    5.000000
50%    2009.000000    9.700000
75%    2016.000000   21.800000
max    2022.000000  151.100000

In [53]:
# Counting 
df8.isnull().sum()

STATISTIC Label     0
Year                0
Sex                 0
Nationality         0
UNIT                0
VALUE              27
dtype: int64

In [54]:
# Display the rows with null values
df8_NaN = df8[df8.isnull().any(axis=1)]
print("Rows with null values:")
df8_NaN.head(10)

Rows with null values:


STATISTIC Label  Year         Sex  \
3   Estimated Immigration (Persons in April)  1996  Both sexes   
9   Estimated Immigration (Persons in April)  1996        Male   
15  Estimated Immigration (Persons in April)  1996      Female   
21  Estimated Immigration (Persons in April)  1997  Both sexes   
27  Estimated Immigration (Persons in April)  1997        Male   
33  Estimated Immigration (Persons in April)  1997      Female   
39  Estimated Immigration (Persons in April)  1998  Both sexes   
45  Estimated Immigration (Persons in April)  1998        Male   
51  Estimated Immigration (Persons in April)  1998      Female   
57  Estimated Immigration (Persons in April)  1999  Both sexes   

                                          Nationality      UNIT  VALUE  
3   EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
9   EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
15  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
21  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
27  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
33  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
39  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
45  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
51  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
57  EU15 to EU27 (accession countries joined post ...  Thousand    NaN

In [55]:
#Function to count the NaN values per category:
def df8_NaN_Count(df8_NaN):
    for column in df8_NaN:
        counts = df8_NaN[column].value_counts()
        print(f"Occurrences for {column}:")
        print(counts)
        print("\n")


# Call the function with your DataFrame as an argument
df8_NaN_Count(df8_NaN)

Occurrences for STATISTIC Label:
Estimated Immigration (Persons in April)    27
Name: STATISTIC Label, dtype: int64


Occurrences for Year:
1996    3
1997    3
1998    3
1999    3
2000    3
2001    3
2002    3
2003    3
2004    3
Name: Year, dtype: int64


Occurrences for Sex:
Both sexes    9
Male          9
Female        9
Name: Sex, dtype: int64


Occurrences for Nationality:
EU15 to EU27 (accession countries joined post 2004)    27
Name: Nationality, dtype: int64


Occurrences for UNIT:
Thousand    27
Name: UNIT, dtype: int64


Occurrences for VALUE:
Series([], Name: VALUE, dtype: int64)




Conclusion:
    
This dataset has 27 NaN values but that is due to one of is Nationality categories, EU15 to EU27 (accession countries joined post 2004). We are missing values for this label years prior to 2004. Total of  486 entries.

## 9. Looking up Estimated Migration (Persons in April), PEA18.20231004T131041.csv:

In [56]:
# Loading the dataset:
df9 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA18.20231004T131041.csv')

In [57]:
# Displaying first 5 rows:
df9.head()

STATISTIC Label  Year             Country  \
0  Estimated Migration (Persons in April)  1987  United Kingdom (1)   
1  Estimated Migration (Persons in April)  1987  United Kingdom (1)   
2  Estimated Migration (Persons in April)  1987  United Kingdom (1)   
3  Estimated Migration (Persons in April)  1987  United Kingdom (1)   
4  Estimated Migration (Persons in April)  1987  United Kingdom (1)   

          Sex        Origin or Destination      UNIT  VALUE  
0  Both sexes                Net migration  Thousand  -13.7  
1  Both sexes  Emigrants: All destinations  Thousand   21.8  
2  Both sexes      Immigrants: All origins  Thousand    8.1  
3        Male                Net migration  Thousand   -9.0  
4        Male  Emigrants: All destinations  Thousand   13.1

In [58]:
# Looking up distinct values within the df:
def df9_distinct(df9):
    for column in df9:
        unique_values = df9[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df9_distinct(df9)

Unique values for STATISTIC Label:
['Estimated Migration (Persons in April)']


Unique values for Year:
[1987 1988 1989 1990 1991 1992 1993 1994 1995 1996 1997 1998 1999 2000
 2001 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014
 2015 2016 2017 2018 2019 2020 2021 2022 2023]


Unique values for Country:
['United Kingdom (1)' 'United States' 'Canada' 'Australia'
 'Other countries (23)' 'All countries' 'EU14 excl Irl (UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)']


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Origin or Destination:
['Net migration' 'Emigrants: All destinations' 'Immigrants: All origins']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[-1.370e+01  2.180e+01  8.100e+00 -9.000e+00  1.310e+01  4.100e+00
 -4.700e+00  8.700e+00  4.000e+00 -6.900e+00  9.900e+00  3.000e+00
 -3.600e+00  4.800e+00  1.200e+00 -3.500e+00  5.200e+00  1.700e+00
        nan -1.400e+00  5.400e+00 -8.000e-01  2.600e

In [59]:
# Getting df info
df9.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2664 entries, 0 to 2663
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   STATISTIC Label        2664 non-null   object 
 1   Year                   2664 non-null   int64  
 2   Country                2664 non-null   object 
 3   Sex                    2664 non-null   object 
 4   Origin or Destination  2664 non-null   object 
 5   UNIT                   2664 non-null   object 
 6   VALUE                  2104 non-null   float64
dtypes: float64(1), int64(1), object(5)
memory usage: 145.8+ KB


In [60]:
# Describing the df:
df9.describe()

Year        VALUE
count  2664.000000  2104.000000
mean   2005.000000     8.943726
std      10.679083    15.513703
min    1987.000000   -43.900000
25%    1996.000000     1.800000
50%    2005.000000     4.700000
75%    2014.000000    10.200000
max    2023.000000   151.100000

In [61]:
# Counting NaN Values
df9.isnull().sum()

STATISTIC Label            0
Year                       0
Country                    0
Sex                        0
Origin or Destination      0
UNIT                       0
VALUE                    560
dtype: int64

In [62]:
# Display the rows with null values
df9_NaN = df9[df9.isnull().any(axis=1)]
print("Rows with null values:")
df9_NaN.head(10)

Rows with null values:


STATISTIC Label  Year    Country         Sex  \
18  Estimated Migration (Persons in April)  1987     Canada  Both sexes   
19  Estimated Migration (Persons in April)  1987     Canada  Both sexes   
20  Estimated Migration (Persons in April)  1987     Canada  Both sexes   
21  Estimated Migration (Persons in April)  1987     Canada        Male   
22  Estimated Migration (Persons in April)  1987     Canada        Male   
23  Estimated Migration (Persons in April)  1987     Canada        Male   
24  Estimated Migration (Persons in April)  1987     Canada      Female   
25  Estimated Migration (Persons in April)  1987     Canada      Female   
26  Estimated Migration (Persons in April)  1987     Canada      Female   
27  Estimated Migration (Persons in April)  1987  Australia  Both sexes   

          Origin or Destination      UNIT  VALUE  
18                Net migration  Thousand    NaN  
19  Emigrants: All destinations  Thousand    NaN  
20      Immigrants: All origins  Thousand    NaN  
21                Net migration  Thousand    NaN  
22  Emigrants: All destinations  Thousand    NaN  
23      Immigrants: All origins  Thousand    NaN  
24                Net migration  Thousand    NaN  
25  Emigrants: All destinations  Thousand    NaN  
26      Immigrants: All origins  Thousand    NaN  
27                Net migration  Thousand    NaN

In [63]:
#Function to count the NaN values per category:
def df9_NaN_Count(df9_NaN):
    for column in df9_NaN:
        counts = df9_NaN[column].value_counts()
        print(f"Occurrences for {column}:")
        print(counts)
        print("\n")


# Call the function with your DataFrame as an argument
df9_NaN_Count(df9_NaN)

Occurrences for STATISTIC Label:
Estimated Migration (Persons in April)    560
Name: STATISTIC Label, dtype: int64


Occurrences for Year:
1987    31
1989    31
1990    31
1991    31
1988    31
1999    27
2004    27
2003    27
2002    27
2001    27
2000    27
1997    27
1998    27
1996    27
1995    27
1994    27
1993    27
1992    27
2005    18
2006    18
2007    18
Name: Year, dtype: int64


Occurrences for Country:
Canada                                                 189
Australia                                              189
EU15 to EU27 (accession countries joined post 2004)    162
All countries                                           20
Name: Country, dtype: int64


Occurrences for Sex:
Male          190
Female        190
Both sexes    180
Name: Sex, dtype: int64


Occurrences for Origin or Destination:
Emigrants: All destinations    190
Immigrants: All origins        190
Net migration                  180
Name: Origin or Destination, dtype: int64


Occurrences for UNIT:
T

Conclusion:
    
It has 560 NaN values and 2664 entries.

## 10. Looking up Estimated Migration Aged 15 Years and Over (Persons in April), PEA19.20231004T131056.csv:

In [64]:
# Loading the dataset:
df10 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA19.20231004T131056.csv')

In [65]:
# Displaying first 5 rows:
df10.head()

STATISTIC Label  Year         Sex  \
0  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
1  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
2  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
3  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
4  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   

              Education Level       Inward or Outward Flow      UNIT  VALUE  
0  Higher secondary and below                Net migration  Thousand   -9.7  
1  Higher secondary and below  Emigrants: All destinations  Thousand   31.6  
2  Higher secondary and below      Immigrants: All origins  Thousand   21.9  
3           Post leaving cert                Net migration  Thousand   -0.2  
4           Post leaving cert  Emigrants: All destinations  Thousand    6.7

In [66]:
# Looking up distinct values within the df:
def df10_distinct(df10):
    for column in df10:
        unique_values = df10[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df10_distinct(df10)

Unique values for STATISTIC Label:
['Estimated Migration Aged 15 Years and Over  (Persons in April)']


Unique values for Year:
[2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Education Level:
['Higher secondary and below' 'Post leaving cert' 'Third level'
 'Not stated' 'All persons']


Unique values for Inward or Outward Flow:
['Net migration' 'Emigrants: All destinations' 'Immigrants: All origins']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[-9.70e+00  3.16e+01  2.19e+01 -2.00e-01  6.70e+00  6.50e+00  8.10e+00
  2.44e+01  3.25e+01  1.30e+00  7.00e+00  8.30e+00 -5.00e-01  6.97e+01
  6.91e+01 -6.90e+00  1.86e+01  1.17e+01 -2.10e+00  5.10e+00  2.90e+00
  2.60e+00  1.36e+01  1.62e+01  1.00e-01  3.60e+00  3.70e+00 -6.20e+00
  4.08e+01  3.46e+01 -2.90e+00  1.30e+01  1.02e+01  2.00e+00  1.60e+00
  5.40e+00  1.08e+01  1.20e+00  3.40e+00  4.60e+00  5.70e+00  2.89e+01
 -1.

In [67]:
# Getting df info
df10.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 630 entries, 0 to 629
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   STATISTIC Label         630 non-null    object 
 1   Year                    630 non-null    int64  
 2   Sex                     630 non-null    object 
 3   Education Level         630 non-null    object 
 4   Inward or Outward Flow  630 non-null    object 
 5   UNIT                    630 non-null    object 
 6   VALUE                   630 non-null    float64
dtypes: float64(1), int64(1), object(5)
memory usage: 34.6+ KB


In [68]:
# Describing the df:
df10.describe()

Year       VALUE
count   630.000000  630.000000
mean   2015.500000   11.867302
std       4.034332   17.222591
min    2009.000000  -28.300000
25%    2012.000000    1.300000
50%    2015.500000    5.550000
75%    2019.000000   17.375000
max    2022.000000  102.000000

Conclusion:
    
No NaN values for this dataset and 630 entries.

## 11. Looking up Estimated Migration Aged 15 Years and Over (Persons in April), PEA20.20231004T131025.csv:

In [69]:
# Loading the dataset:
df11 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA20.20231004T131025.csv')

In [70]:
# Displaying first 5 rows:
df11.head()

STATISTIC Label  Year  \
0  Estimated Migration Aged 15 Years and Over (Pe...  2009   
1  Estimated Migration Aged 15 Years and Over (Pe...  2009   
2  Estimated Migration Aged 15 Years and Over (Pe...  2009   
3  Estimated Migration Aged 15 Years and Over (Pe...  2009   
4  Estimated Migration Aged 15 Years and Over (Pe...  2009   

  Principal Economic Status         Sex       Inward or Outward Flow  \
0               All persons  Both sexes                Net migration   
1               All persons  Both sexes  Emigrants: All destinations   
2               All persons  Both sexes      Immigrants: All origins   
3               All persons        Male                Net migration   
4               All persons        Male  Emigrants: All destinations   

       UNIT  VALUE  
0  Thousand   -0.5  
1  Thousand   69.7  
2  Thousand   69.1  
3  Thousand   -6.2  
4  Thousand   40.8

In [71]:
# Looking up distinct values within the df:
def df11_distinct(df11):
    for column in df11:
        unique_values = df11[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df11_distinct(df11)

Unique values for STATISTIC Label:
['Estimated Migration Aged 15 Years and Over (Persons in April)']


Unique values for Year:
[2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022]


Unique values for Principal Economic Status:
['All persons' 'Persons at work' 'All unemployed persons'
 'Student or pupil' 'Other economic status']


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Inward or Outward Flow:
['Net migration' 'Emigrants: All destinations' 'Immigrants: All origins']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[-5.00e-01  6.97e+01  6.91e+01 -6.20e+00  4.08e+01  3.46e+01  5.70e+00
  2.89e+01 -7.60e+00  4.13e+01  3.37e+01 -6.80e+00  2.56e+01  1.88e+01
 -8.00e-01  1.57e+01  1.49e+01  4.80e+00  1.08e+01  2.20e+00  7.70e+00
  9.80e+00  2.60e+00  3.20e+00  5.80e+00 -2.70e+00  1.25e+01 -2.10e+00
  6.00e+00  3.80e+00 -6.00e-01  6.50e+00  5.90e+00  5.00e+00  5.10e+00
  1.01e+01  5.00e-01  1.50e+00  2.10e+00  4.40e+00  3.50

In [72]:
# Getting df info
df11.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 630 entries, 0 to 629
Data columns (total 7 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   STATISTIC Label            630 non-null    object 
 1   Year                       630 non-null    int64  
 2   Principal Economic Status  630 non-null    object 
 3   Sex                        630 non-null    object 
 4   Inward or Outward Flow     630 non-null    object 
 5   UNIT                       630 non-null    object 
 6   VALUE                      630 non-null    float64
dtypes: float64(1), int64(1), object(5)
memory usage: 34.6+ KB


In [73]:
# Describing the df:
df11.describe()

Year       VALUE
count   630.000000  630.000000
mean   2015.500000   11.867619
std       4.034332   16.670281
min    2009.000000  -28.200000
25%    2012.000000    2.225000
50%    2015.500000    6.750000
75%    2019.000000   16.475000
max    2022.000000  102.000000

Conclusion:

Datasets df10 with df11 have the same attributes the only difference is the Education Level (df10) and Economic Status.
That determines different migratory values. Both datasets have the same entries.

## 12. Estimated Population, PEA21.20231004T131004.csv:

In [74]:
# Loading the dataset:
df12 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA21.20231004T131004.csv')

In [75]:
# Displaying first 5 rows:
df12.head()

STATISTIC Label  Year         Sex  \
0  Estimated Population  2006  Both sexes   
1  Estimated Population  2006  Both sexes   
2  Estimated Population  2006  Both sexes   
3  Estimated Population  2006  Both sexes   
4  Estimated Population  2006  Both sexes   

                                         Nationality      UNIT   VALUE  
0                                              Irish  Thousand  3802.4  
1                                             UK (1)  Thousand   115.5  
2  EU14 excl Irl (countries in the EU pre 2004 ex...  Thousand    43.8  
3  EU15 to EU27 (accession countries joined post ...  Thousand   132.5  
4                            Other nationalities(17)  Thousand   138.8

In [76]:
# Looking up distinct values within the df:
def df12_distinct(df12):
    for column in df12:
        unique_values = df12[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df12_distinct(df12)

Unique values for STATISTIC Label:
['Estimated Population']


Unique values for Year:
[2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019
 2020 2021 2022]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Nationality:
['Irish' 'UK (1)'
 'EU14 excl Irl (countries in the EU pre 2004 excluding UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)'
 'Other nationalities(17)' 'All nationalities']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[3802.4  115.5   43.8  132.5  138.8 4232.9 1887.4   57.8   21.3   79.9
   70.9 2117.3 1914.9   57.7   22.5   52.6   67.9 2115.6 3856.2   47.
  207.6  149.6 4375.8 1915.7   57.6   21.8  120.3   75.9 2191.3 1940.5
   57.9   25.2   87.4   73.7 2184.6 3909.5  117.9   50.8  247.7  159.2
 4485.1 1942.3   58.9   23.7  133.9   79.8 2238.6 1967.2   59.    27.1
  113.8   79.5 2246.5 3958.   117.1   55.3  240.8  162.2 4533.4 1966.4
   58.3   25.5  124.8   82.3 2257.3 1991.6   58.8   29

In [77]:
# Getting df info
df12.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 306 entries, 0 to 305
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  306 non-null    object 
 1   Year             306 non-null    int64  
 2   Sex              306 non-null    object 
 3   Nationality      306 non-null    object 
 4   UNIT             306 non-null    object 
 5   VALUE            306 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 14.5+ KB


In [78]:
# Describing the df:
df12.describe()

Year        VALUE
count   306.000000   306.000000
mean   2014.000000  1041.806536
std       4.907004  1477.104974
min    2006.000000    21.300000
25%    2010.000000    60.100000
50%    2014.000000   121.400000
75%    2018.000000  2117.600000
max    2022.000000  5100.200000

Conclusion:
    
No NaN values and 306 entries.

## 13. Estimated Emigration, PEA23.20231004T131045.csv:

In [79]:
# Loading the dataset:
df13 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA23.20231004T131045.csv')

In [80]:
# Displaying first 5 rows:
df13.head()

Statistic Label  Year         Sex  \
0  Estimated Emigration (Persons in April)  2006  Both sexes   
1  Estimated Emigration (Persons in April)  2006  Both sexes   
2  Estimated Emigration (Persons in April)  2006  Both sexes   
3  Estimated Emigration (Persons in April)  2006  Both sexes   
4  Estimated Emigration (Persons in April)  2006  Both sexes   

                                         Citizenship      UNIT  VALUE  
0                                      All Countries  Thousand   36.0  
1  EU14 excl Irl (countries in the EU pre 2004 ex...  Thousand    5.1  
2  EU15 to EU27 (accession countries joined post ...  Thousand    7.2  
3                                     United Kingdom  Thousand    2.2  
4                                            Ireland  Thousand   15.3

In [81]:
# Looking up distinct values within the df:
def df13_distinct(df13):
    for column in df13:
        unique_values = df13[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df13_distinct(df13)

Unique values for Statistic Label:
['Estimated Emigration (Persons in April)']


Unique values for Year:
[2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019
 2020 2021 2022 2023]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Citizenship:
['All Countries'
 'EU14 excl Irl (countries in the EU pre 2004 excluding UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)' 'United Kingdom'
 'Ireland' 'All countries excluding Ireland,United Kingdom and EU272020']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[36.   5.1  7.2  2.2 15.3  6.2 18.7  2.5  3.7  1.6  8.   2.8 17.3  2.6
  3.4  0.6  7.3 46.3  8.9 12.6 12.9  8.2 25.7  4.3  8.4  2.3  6.1  4.6
 20.6  1.4  6.8  3.6 49.2  6.  17.2 13.1  9.  29.6  2.  12.4  7.1  5.5
 19.6  4.   4.8  1.3  3.5 72.   7.4 30.5  3.9 19.2 11.  41.9  1.9 21.1
  2.4 12.3 30.1  9.5  1.5  6.9  6.7 69.2 19.   3.  28.9  9.3 40.6 12.8
  1.7 16.7  5.4 28.7  5.  12.2 80.6 10.2 13.9 42.   9.9  1

In [82]:
# Getting df info
df13.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 324 entries, 0 to 323
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Statistic Label  324 non-null    object 
 1   Year             324 non-null    int64  
 2   Sex              324 non-null    object 
 3   Citizenship      324 non-null    object 
 4   UNIT             324 non-null    object 
 5   VALUE            324 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 15.3+ KB


In [83]:
# Describing the df:
df13.describe()

Year       VALUE
count   324.000000  324.000000
mean   2014.500000   13.737963
std       5.196152   16.095976
min    2006.000000    0.600000
25%    2010.000000    3.600000
50%    2014.500000    6.700000
75%    2019.000000   17.800000
max    2023.000000   83.000000

Conclusion:

No NaN values and 324 entries.

## 14. Estimated Immigration, PEA24.20231004T131048.csv:

In [84]:
# Loading the dataset:
df14 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA24.20231004T131048.csv')

In [85]:
# Displaying first 5 rows:
df14.head()

Statistic Label  Year         Sex  \
0  Estimated Immigration (Persons in April)  1996  Both sexes   
1  Estimated Immigration (Persons in April)  1996  Both sexes   
2  Estimated Immigration (Persons in April)  1996  Both sexes   
3  Estimated Immigration (Persons in April)  1996  Both sexes   
4  Estimated Immigration (Persons in April)  1996  Both sexes   

                                         Citizenship      UNIT  VALUE  
0                                      All Countries  Thousand   39.2  
1  EU14 excl Irl (countries in the EU pre 2004 ex...  Thousand    5.0  
2  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
3                                     United Kingdom  Thousand    8.3  
4                                            Ireland  Thousand   17.7

In [86]:
# Looking up distinct values within the df:
def df14_distinct(df14):
    for column in df14:
        unique_values = df14[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df14_distinct(df14)

Unique values for Statistic Label:
['Estimated Immigration (Persons in April)']


Unique values for Year:
[1996 1997 1998 1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009
 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Citizenship:
['All Countries'
 'EU14 excl Irl (countries in the EU pre 2004 excluding UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)' 'United Kingdom'
 'Ireland' 'All countries excluding Ireland,United Kingdom and EU272020']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[ 39.2   5.    nan   8.3  17.7   8.2  18.8   2.2   4.3   4.   20.4   2.8
   3.9   9.5   4.2  44.5   5.5   8.4  20.8   9.7  22.    2.5   4.6  10.2
   4.7  22.6   3.1   3.8  10.7   5.1  46.    6.1   8.6  24.3   7.   22.7
   2.3  12.3   3.7  23.3   4.4  11.9   3.3  48.9   6.9  26.7  24.5   3.
  13.7   3.6  24.4  13.    3.5  52.6  24.8  11.1  26.2   4.1  11.8   6.

In [87]:
# Getting df info
df14.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 504 entries, 0 to 503
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Statistic Label  504 non-null    object 
 1   Year             504 non-null    int64  
 2   Sex              504 non-null    object 
 3   Citizenship      504 non-null    object 
 4   UNIT             504 non-null    object 
 5   VALUE            477 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 23.8+ KB


In [88]:
# Describing the df:
df14.describe()

Year       VALUE
count   504.000000  477.000000
mean   2009.500000   18.060168
std       8.085773   21.330315
min    1996.000000    1.100000
25%    2002.750000    5.000000
50%    2009.500000    9.700000
75%    2016.250000   22.600000
max    2023.000000  151.100000

In [89]:
# Counting NaN Values
df14.isnull().sum()

Statistic Label     0
Year                0
Sex                 0
Citizenship         0
UNIT                0
VALUE              27
dtype: int64

In [90]:
# Display the rows with null values
df14_NaN = df14[df14.isnull().any(axis=1)]
print("Rows with null values:")
df14_NaN.head(10)

Rows with null values:


Statistic Label  Year         Sex  \
2   Estimated Immigration (Persons in April)  1996  Both sexes   
8   Estimated Immigration (Persons in April)  1996        Male   
14  Estimated Immigration (Persons in April)  1996      Female   
20  Estimated Immigration (Persons in April)  1997  Both sexes   
26  Estimated Immigration (Persons in April)  1997        Male   
32  Estimated Immigration (Persons in April)  1997      Female   
38  Estimated Immigration (Persons in April)  1998  Both sexes   
44  Estimated Immigration (Persons in April)  1998        Male   
50  Estimated Immigration (Persons in April)  1998      Female   
56  Estimated Immigration (Persons in April)  1999  Both sexes   

                                          Citizenship      UNIT  VALUE  
2   EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
8   EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
14  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
20  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
26  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
32  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
38  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
44  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
50  EU15 to EU27 (accession countries joined post ...  Thousand    NaN  
56  EU15 to EU27 (accession countries joined post ...  Thousand    NaN

In [91]:
#Function to count the NaN values per category:
def df14_NaN_Count(df14_NaN):
    for column in df14_NaN:
        counts = df14_NaN[column].value_counts()
        print(f"Occurrences for {column}:")
        print(counts)
        print("\n")


# Call the function with your DataFrame as an argument
df14_NaN_Count(df14_NaN)

Occurrences for Statistic Label:
Estimated Immigration (Persons in April)    27
Name: Statistic Label, dtype: int64


Occurrences for Year:
1996    3
1997    3
1998    3
1999    3
2000    3
2001    3
2002    3
2003    3
2004    3
Name: Year, dtype: int64


Occurrences for Sex:
Both sexes    9
Male          9
Female        9
Name: Sex, dtype: int64


Occurrences for Citizenship:
EU15 to EU27 (accession countries joined post 2004)    27
Name: Citizenship, dtype: int64


Occurrences for UNIT:
Thousand    27
Name: UNIT, dtype: int64


Occurrences for VALUE:
Series([], Name: VALUE, dtype: int64)




Conclusion:
    
This dataset has 27 NaN values but that is due to one of is Nationality categories, EU15 to EU27 (accession countries joined post 2004). We are missing values for this label years prior to 2004. Total of  504 entries.
Very similar dataset to PEA17.20231004T131052.csv.

## 15. Estimated Population, PEA25.20231004T131052.csv:

In [92]:
# Loading the dataset:
df15 = pd.read_csv('..\\Datasets_data.cso.ie\\PEA25.20231004T131052.csv')

In [93]:
# Displaying first 5 rows:
df15.head()

Statistic Label  Year         Sex  \
0  Estimated Population  2006  Both sexes   
1  Estimated Population  2006  Both sexes   
2  Estimated Population  2006  Both sexes   
3  Estimated Population  2006  Both sexes   
4  Estimated Population  2006  Both sexes   

                                         Citizenship      UNIT   VALUE  
0                                      All Countries  Thousand  4232.9  
1  EU14 excl Irl (countries in the EU pre 2004 ex...  Thousand    43.8  
2  EU15 to EU27 (accession countries joined post ...  Thousand   132.5  
3                                     United Kingdom  Thousand   115.5  
4                                            Ireland  Thousand  3802.4

In [94]:
# Looking up distinct values within the df:
def df15_distinct(df15):
    for column in df15:
        unique_values = df15[column].unique()
        print(f"Unique values for {column}:")
        print(unique_values)
        print("\n")

# Call the function with your DataFrame as an argument
df15_distinct(df15)

Unique values for Statistic Label:
['Estimated Population']


Unique values for Year:
[2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019
 2020 2021 2022 2023]


Unique values for Sex:
['Both sexes' 'Male' 'Female']


Unique values for Citizenship:
['All Countries'
 'EU14 excl Irl (countries in the EU pre 2004 excluding UK & Ireland)'
 'EU15 to EU27 (accession countries joined post 2004)' 'United Kingdom'
 'Ireland' 'All countries excluding Ireland,United Kingdom and EU272020']


Unique values for UNIT:
['Thousand']


Unique values for VALUE:
[4232.9   43.8  132.5  115.5 3802.4  138.8 2117.3   21.3   79.9   57.8
 1887.4   70.9 2115.6   22.5   52.6   57.7 1914.9   67.9 4375.8   47.
  207.6 3856.2  149.6 2191.3   21.8  120.3   57.6 1915.7   75.9 2184.6
   25.2   87.4   57.9 1940.5   73.7 4485.1   50.8  247.7  117.9 3909.5
  159.2 2238.6   23.7  133.9   58.9 1942.3   79.8 2246.5   27.1  113.8
   59.  1967.2   79.5 4533.4   55.3  240.8  117.1 3958.   162.2 2257.3
   25.5

In [95]:
# Getting df info
df15.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 324 entries, 0 to 323
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Statistic Label  324 non-null    object 
 1   Year             324 non-null    int64  
 2   Sex              324 non-null    object 
 3   Citizenship      324 non-null    object 
 4   UNIT             324 non-null    object 
 5   VALUE            324 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 15.3+ KB


In [96]:
# Describing the df:
df15.describe()

Year        VALUE
count   324.000000   324.000000
mean   2014.500000  1052.622531
std       5.196152  1492.635331
min    2006.000000    21.300000
25%    2010.000000    58.975000
50%    2014.500000   121.350000
75%    2019.000000  2145.400000
max    2023.000000  5281.600000

Conclusion:
    
No NaN values and 324 entries, very similar to dataset PEA21.20231004T131004.csv (df12).

# Summary:

## 7 Datasets regarding Population:

 1.Looking up Population Estimates (Persons in April), PEA01.20231004T131011.csv (df1):
 
 3.Looking up Estimated Population (Persons in April), PEA04.20231004T131011.csv (df3):
 
 4.Looking up Estimated Population (Persons in April), PEA07.20231004T131011.csv (df4):
 
 5.Looking up Population estimates from 1926, PEA11.20231004T131042.csv (df5):
 
 6.Looking up Anual Population Change, PEA15.20231004T131032.csv (df6):
 
 12.Estimated Population, PEA21.20231004T131004.csv (df12):
 
 15.Estimated Population, PEA25.20231004T131052.csv (df15):
 
 
## 4 Datasets regarding Migration:

 2.Looking up Estimated Migration (Persons in April) PEA03.20231004T131049.csv (df2):
 
 9.Looking up Estimated Migration (Persons in April), PEA18.20231004T131041.csv (df9):
 
 10.Looking up Estimated Migration Aged 15 Years and Over (Persons in April), PEA19.20231004T131056.csv (df10):
 
 11.Looking up Estimated Migration Aged 15 Years and Over (Persons in April), PEA20.20231004T131025.csv (df11):
 

 
 ## 3 Datasets regarding Emigration:
 
 7.Looking up Estimated Emigration (Persons in April), PEA16.20231004T131017.csv (df7):
 
 8.Looking up Estimated Emigration (Persons in April), PEA17.20231004T131052.csv (df8):
 
 13.Estimated Emigration, PEA23.20231004T131045.csv (df13):

 ## 1 Dataset regarding Immigration:
 
 
 14.Estimated Immigration, PEA24.20231004T131048.csv (df14):


In [97]:
# Getting df info
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5994 entries, 0 to 5993
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  5994 non-null   object 
 1   Year             5994 non-null   int64  
 2   Age Group        5994 non-null   object 
 3   Sex              5994 non-null   object 
 4   UNIT             5994 non-null   object 
 5   VALUE            5880 non-null   float64
dtypes: float64(1), int64(1), object(4)
memory usage: 281.1+ KB


In [98]:
# Getting df info
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1998 entries, 0 to 1997
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   STATISTIC Label         1998 non-null   object 
 1   Year                    1998 non-null   int64  
 2   Age Group               1998 non-null   object 
 3   Sex                     1998 non-null   object 
 4   Inward or Outward Flow  1998 non-null   object 
 5   UNIT                    1998 non-null   object 
 6   VALUE                   1683 non-null   float64
dtypes: float64(1), int64(1), object(5)
memory usage: 109.4+ KB


In [99]:
# Getting df info
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6669 entries, 0 to 6668
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  6669 non-null   object 
 1   Year             6669 non-null   int64  
 2   Age Group        6669 non-null   object 
 3   Sex              6669 non-null   object 
 4   Region           6669 non-null   object 
 5   UNIT             6669 non-null   object 
 6   VALUE            6669 non-null   float64
dtypes: float64(1), int64(1), object(5)
memory usage: 364.8+ KB


In [100]:
# Getting df info
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11286 entries, 0 to 11285
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Statistic Label          11286 non-null  object 
 1   Age Group                11286 non-null  object 
 2   Sex                      11286 non-null  object 
 3   Regional Authority Area  11286 non-null  object 
 4   Year                     11286 non-null  int64  
 5   UNIT                     11286 non-null  object 
 6   VALUE                    11286 non-null  float64
dtypes: float64(1), int64(1), object(5)
memory usage: 617.3+ KB


In [101]:
# Getting df info
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11817 entries, 0 to 11816
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   STATISTIC Label     11817 non-null  object
 1   Year                11817 non-null  int64 
 2   Single Year of Age  11817 non-null  object
 3   Sex                 11817 non-null  object
 4   UNIT                11817 non-null  object
 5   VALUE               11817 non-null  int64 
dtypes: int64(2), object(4)
memory usage: 554.0+ KB


In [102]:
# Getting df info
df6.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 584 entries, 0 to 583
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  584 non-null    object 
 1   Year             584 non-null    int64  
 2   Component        584 non-null    object 
 3   UNIT             584 non-null    object 
 4   VALUE            440 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 22.9+ KB


In [103]:
# Getting df info
df7.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 306 entries, 0 to 305
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  306 non-null    object 
 1   Year             306 non-null    int64  
 2   Sex              306 non-null    object 
 3   Nationality      306 non-null    object 
 4   UNIT             306 non-null    object 
 5   VALUE            306 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 14.5+ KB


In [104]:
# Getting df info
df8.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 486 entries, 0 to 485
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  486 non-null    object 
 1   Year             486 non-null    int64  
 2   Sex              486 non-null    object 
 3   Nationality      486 non-null    object 
 4   UNIT             486 non-null    object 
 5   VALUE            459 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 22.9+ KB


In [105]:
# Getting df info
df9.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2664 entries, 0 to 2663
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   STATISTIC Label        2664 non-null   object 
 1   Year                   2664 non-null   int64  
 2   Country                2664 non-null   object 
 3   Sex                    2664 non-null   object 
 4   Origin or Destination  2664 non-null   object 
 5   UNIT                   2664 non-null   object 
 6   VALUE                  2104 non-null   float64
dtypes: float64(1), int64(1), object(5)
memory usage: 145.8+ KB


In [106]:
# Getting df info
df10.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 630 entries, 0 to 629
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   STATISTIC Label         630 non-null    object 
 1   Year                    630 non-null    int64  
 2   Sex                     630 non-null    object 
 3   Education Level         630 non-null    object 
 4   Inward or Outward Flow  630 non-null    object 
 5   UNIT                    630 non-null    object 
 6   VALUE                   630 non-null    float64
dtypes: float64(1), int64(1), object(5)
memory usage: 34.6+ KB


In [107]:
# Getting df info
df11.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 630 entries, 0 to 629
Data columns (total 7 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   STATISTIC Label            630 non-null    object 
 1   Year                       630 non-null    int64  
 2   Principal Economic Status  630 non-null    object 
 3   Sex                        630 non-null    object 
 4   Inward or Outward Flow     630 non-null    object 
 5   UNIT                       630 non-null    object 
 6   VALUE                      630 non-null    float64
dtypes: float64(1), int64(1), object(5)
memory usage: 34.6+ KB


In [108]:
# Getting df info
df12.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 306 entries, 0 to 305
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   STATISTIC Label  306 non-null    object 
 1   Year             306 non-null    int64  
 2   Sex              306 non-null    object 
 3   Nationality      306 non-null    object 
 4   UNIT             306 non-null    object 
 5   VALUE            306 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 14.5+ KB


In [109]:
# Getting df info
df13.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 324 entries, 0 to 323
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Statistic Label  324 non-null    object 
 1   Year             324 non-null    int64  
 2   Sex              324 non-null    object 
 3   Citizenship      324 non-null    object 
 4   UNIT             324 non-null    object 
 5   VALUE            324 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 15.3+ KB


In [110]:
# Getting df info
df14.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 504 entries, 0 to 503
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Statistic Label  504 non-null    object 
 1   Year             504 non-null    int64  
 2   Sex              504 non-null    object 
 3   Citizenship      504 non-null    object 
 4   UNIT             504 non-null    object 
 5   VALUE            477 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 23.8+ KB


In [111]:
# Getting df info
df15.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 324 entries, 0 to 323
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Statistic Label  324 non-null    object 
 1   Year             324 non-null    int64  
 2   Sex              324 non-null    object 
 3   Citizenship      324 non-null    object 
 4   UNIT             324 non-null    object 
 5   VALUE            324 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 15.3+ KB


In [112]:
df10.head()

STATISTIC Label  Year         Sex  \
0  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
1  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
2  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
3  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   
4  Estimated Migration Aged 15 Years and Over  (P...  2009  Both sexes   

              Education Level       Inward or Outward Flow      UNIT  VALUE  
0  Higher secondary and below                Net migration  Thousand   -9.7  
1  Higher secondary and below  Emigrants: All destinations  Thousand   31.6  
2  Higher secondary and below      Immigrants: All origins  Thousand   21.9  
3           Post leaving cert                Net migration  Thousand   -0.2  
4           Post leaving cert  Emigrants: All destinations  Thousand    6.7

In [113]:
df11.head()

STATISTIC Label  Year  \
0  Estimated Migration Aged 15 Years and Over (Pe...  2009   
1  Estimated Migration Aged 15 Years and Over (Pe...  2009   
2  Estimated Migration Aged 15 Years and Over (Pe...  2009   
3  Estimated Migration Aged 15 Years and Over (Pe...  2009   
4  Estimated Migration Aged 15 Years and Over (Pe...  2009   

  Principal Economic Status         Sex       Inward or Outward Flow  \
0               All persons  Both sexes                Net migration   
1               All persons  Both sexes  Emigrants: All destinations   
2               All persons  Both sexes      Immigrants: All origins   
3               All persons        Male                Net migration   
4               All persons        Male  Emigrants: All destinations   

       UNIT  VALUE  
0  Thousand   -0.5  
1  Thousand   69.7  
2  Thousand   69.1  
3  Thousand   -6.2  
4  Thousand   40.8